File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_327_A.go
package main

import (
	"fmt"
	"strings"
)

func find(s string, ch string) []int {
	var indices []int
	for i, ltr := range s {
		if string(ltr) == ch {
			indices = append(indices, i)
		}
	}
	return indices
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	var n int
	fmt.Scan(&n)
	var s string
	fmt.Scan(&s)
	s = strings.Replace(s, "", "", -1)
	if !strings.Contains(s, "0") {
		fmt.Println(n - 1)
	} else {
		indices := find(s, "0")
		if len(indices) == 1 {
			fmt.Println(n)
		} else {
			maximum := 0
			for i := 0; i < len(indices); i++ {
				for j := i + 1; j < len(indices); j++ {
					maximum = max(maximum, 2+2*(abs(i-j))-1)-(abs(indices[i]-indices[j])-1))
				}
			}
			fmt.Println(strings.Count(s, "1") + maximum)
		}
	}
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

// End of Code

File: LLM-Translations/GPT-4/avatar/Python/Java/codeforces_327_A.java

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class codeforces_327_A {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine();
        String s = scanner.nextLine().replace("", "");

        if (!s.contains("0")) {
            System.out.println(n - 1);
        } else {
            List<Integer> indices = find(s, '0');
            if (indices.size() == 1) {
                System.out.println(n);
            } else {
                int maximum = 0;
                for (int i = 0; i < indices.size(); i++) {
                    for (int j = i + 1; j < indices.size(); j++) {
                        int index1 = indices.get(i);
                        int index2 = indices.get(j);
                        maximum = Math.max(maximum, 2 + 2 * (Math.abs(i - j) - 1) - (Math.abs(index1 - index2) - 1));
                    }
                }
                System.out.println(s.chars().filter(ch -> ch == '1').count() + maximum);
            }
        }
    }

    private static List<Integer> find(String s, char ch) {
        List<Integer> indices = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ch) {
                indices.add(i);
            }
        }
        return indices;
    }
}

// End of Code.
